include/bluetooth/audio/mics.h

```diff
@@ -0,0 +1,367 @@
+#define BT_MICS_ERR_VAL_OUT_OF_RANGE               0x81
+
+/** Microphone Input Control Service mute states */
+#define BT_MICS_MUTE_UNMUTED                       0x00
+#define BT_MICS_MUTE_MUTED                         0x01
+#define BT_MICS_MUTE_DISABLED                      0x02
+
+/** @brief Register parameters structure for Microphone Input Control Service */
+struct bt_mics_register_param {
+	/** Register parameter structure for Audio Input Control Services */
+	struct bt_aics_register_param aics_param[BT_MICS_AICS_CNT];
+
+	/** Microphone Input Control Service callback structure. */
+	struct bt_mics_cb *cb;
+};
+
+/**
+ * @brief Microphone Input Control Service instance
+ *
+ * Used for to represent a Microphone Input Control Service, for either
+ * a client or a server instance. The instance pointers either represent local
+ * server instances, or remote service instances.
+ */
+struct bt_mics {
+	/** Number of Audio Input Control Service instances */
+	uint8_t aics_cnt;
+	/** Array of pointers to Audio Input Control Service instances */
+	struct bt_aics **aics;
+};
+
+/**
+ * @brief Initialize the Microphone Input Control Service
+ *
+ * This will enable the service and make it discoverable by clients.
+ *
+ * @param param  Pointer to a initialization structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_mics_register(struct bt_mics_register_param *param);
+
+/**
+ * @brief Get Microphone Input Control Service pointer
+ *
+ * Returns a pointer to a struct that contains information about the
+ * Microphone Input Control Service instance, such as pointers to the
+ * Audio Input Control Service instances.
+ *
+ * @param conn          Connection to peer device, or NULL to get server value.
+ * @param[out] service  Pointer to store the result in.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_mics_get(struct bt_conn *conn, struct bt_mics *service);
+
+/**
+ * @brief Callback function for @ref bt_mics_discover.
+ *
+ * This callback is only used for the client.
+ *
+ * @param conn         The connection that was used to discover
+ *                     Microphone Input Control Service.
+ * @param err          Error value. 0 on success, GATT error or errno on fail.
+ * @param aics_count   Number of Audio Input Control Service instances on
+ *                     peer device.
+ */
+typedef void (*bt_mics_discover_cb)(struct bt_conn *conn, int err,
+				    uint8_t aics_count);
+
+/**
+ * @brief Callback function for Microphone Input Control Service mute.
+ *
+ * Called when the value is read,
+ * or if the value is changed by either the server or client.
+ *
+ * @param conn     Connection to peer device, or NULL if local server read.
+ * @param err      Error value. 0 on success, GATT error or errno on fail.
+ *                 For notifications, this will always be 0. For reads, if
+ *                 this is 0, the @p volume and @p mute values are the last
+ *                 known values.
```

Comment:
Will one sometimes get a value that is not the last known value?
Or does this mean that one may get a value, but that there exists newer values (which it is not known what are)?

Reply:
That last part is actually no longer valid. Removed